backend/src/worker_flow.rs

```diff
@@ -91,11 +91,15 @@ pub async fn update_flow_status_after_job_completion(
             Error::InternalErr(format!("requiring status to be parsabled as FlowStatus"))
         })?;
 
+    let skip_loop_failures = skip_loop_failures(flow, old_status.step, &mut tx)
+        .await?
+        .unwrap_or(false);
+
     let (step_counter, new_status) = match &old_status.modules[old_status.step as usize] {
         module_status @ FlowStatusModule::InProgress {
             iterator: Some(Iterator { index, itered, .. }),
             ..
-        } if (index.to_owned() as usize) < itered.len() - 1 && success => {
+        } if (index.to_owned() as usize) < itered.len() - 1 && (success || skip_loop_failures) => {
```

Comment:
I think you might be missing the case where it's the last iteration and that iteration failed, in which case you want to have a consistent behavior and still consider the module to be a success no matter what, which you need to adjust at line 113